src/backend/distributed/commands/table.c

```diff
@@ -219,13 +219,17 @@ PostprocessCreateTableStmtPartitionOf(CreateStmt *createStatement, const
 	if (IsCitusTable(parentRelationId))
 	{
 		Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock, missingOk);
-		Var *parentDistributionColumn = DistPartitionKeyOrError(parentRelationId);
-		char parentDistributionMethod = DISTRIBUTE_BY_HASH;
-		char *parentRelationName = generate_qualified_relation_name(parentRelationId);
-		bool viaDeprecatedAPI = false;
 
-		if (!(IsCitusTable(relationId) && (createStatement->if_not_exists)))
+		/*
+		 * If the relation already exists and it's an IF NOT EXISTS stmt,
+		 * just skip like in plain PG.
+		 */
+		if (!(RelationExists(relationId) && (createStatement->if_not_exists)))
```

Comment:
This causes not distributing new created tables. I mean, for a valid create table command, it gets created after Preprocess function and before Postprocess function. It doesn't matter it's an already existing table or not, `RelationExists(relationId)` will always return true. And when `if_not_exists` is also true, the if condition will not be satisfied, hence the table in a valid creation command will not get distributed.
Example:
```
CREATE TABLE "test2" (
   "cdc" INTEGER,
   "product" INTEGER)
PARTITION BY LIST ("cdc");
SELECT create_distributed_table('test2', 'product');
CREATE TABLE IF NOT EXISTS "test3" PARTITION OF "test2" FOR VALUES IN (1); 
```
Then you can see by connecting to a worker with `psql -p 9701` that it's not distributed

Reply:
Makes sense, I forgot about that 👍

Reply:
The problem here is, when we run `CREATE TABLE IF NOT EXISTS "test3" PARTITION OF "test2" FOR VALUES IN (1);` either we already have a table named test3, or we don't; there will be a table named `test3` when it comes to your if statement. So we couldn't know it's an already existing table or the one we just created (Between Preprocess and Postprocess functions). This is the reason for `(RelationExists(relationId)` is pointless. It will always return true.

Reply:
Yes, I understood what you meant in the first time, I just forgot about that :)

> Then you can see by connecting to a worker with psql -p 9701 that it's not distributed

I was saying for this one, I can check directly from coordinator in pg_dist_partition - test3 will not be there in your example case.